testing/containers/ceph/Dockerfile

```diff
@@ -29,7 +29,7 @@ RUN true \
     && yum clean all \
     && true
 
-ARG GO_VERSION=1.21.8
+ARG GO_VERSION=1.23.7
```

Comment:
maybe we want to force that `ARG` in the future? Go is moving fast, so we will always be updating this. (We are anyway using the go_version value, right?)

Reply:
It's a thought, but I don't like having to do it on the CLI every time. Maybe if the makefile handled it all internally?